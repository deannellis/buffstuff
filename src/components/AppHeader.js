import React from 'react';
import PropTypes from 'prop-types';
import { HamburgerIcon } from '../utils/icons';

const AppHeader = ({ toggleMenu, pageHasMenu }) => {
	return (
		<div
			className={pageHasMenu ? 'app-header' : 'app-header app-header--no-menu'}
		>
			<button
				type="button"
				className="app-header__hamburger-button"
				onClick={() => {
					toggleMenu();
				}}
			>
				<HamburgerIcon size={24} />
			</button>
			<h1>flext</h1>
		</div>
	);
};
AppHeader.propTypes = {
	toggleMenu: PropTypes.func.isRequired,
	pageHasMenu: PropTypes.bool
};
AppHeader.defaultProps = {
	pageHasMenu: true
};

export default AppHeader;
