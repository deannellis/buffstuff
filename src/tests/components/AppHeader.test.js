import React from 'react';
import { shallow } from 'enzyme';

import AppHeader from '../../components/AppHeader';

test('should render App Header with props', () => {
	const wrapper = shallow(<AppHeader pageHasMenu toggleMenu={() => {}} />);
	expect(wrapper).toMatchSnapshot();
});
