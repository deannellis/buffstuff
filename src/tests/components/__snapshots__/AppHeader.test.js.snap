// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App Header with props 1`] = `
<div
  className="app-header"
>
  <button
    className="app-header__hamburger-button"
    onClick={[Function]}
    type="button"
  >
    <HamburgerIcon
      fill="#fff"
      size={24}
    />
  </button>
  <h1>
    flext
  </h1>
</div>
`;
